File: 1-easy/problems/2511-maximum-enemy-forts-that-can-be-captured/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def captureForts(self, forts):
        """
        :type forts: List[int]
        :rtype: int
        """
        result = left = 0
        for right in range(len(forts)):
            if not forts[right]:
                continue
            if forts[right] == -forts[left]:
                result = max(result, right-left-1)
            left = right
        return result

File: 1-easy/problems/2511-maximum-enemy-forts-that-can-be-captured/solution.ts
function captureForts(forts: any): boolean | number | string | any {
  """
  :type forts: List[int]
  :rtype: int
  """
  result = left = 0
  for (right in Array.from({length: (forts}, (_, i) => i).length)) {
      if (not forts[right]) {
          continue
      if (forts[right] == -forts[left]) {
          result = max(result, right-left-1)
      left = right
  return result;
}

export { captureForts };
